### DISCUSSION TITLE
Cassandra/ScyllaDB Database Support

### DISCUSSION THREAD

**Discussion by atsetilam** [RamiAwar/dataline#281] [Ideas]
Hi guys, Dataline seems really interesting. Do you plan on supporting Cassandra/ScyllaDB?

**Comment by RamiAwar** [comment]
Just looked into it, and it seems to be easy but not trivial.

Right now we're using SQLAlchemy to build these connections to DBs, but the plan was always to move to a more abstract representation (to support NoSQL DBs for example, and other query languages like CQL).

The idea would be to replace some of our code on connection creation, on connection schema extraction, etc. with a generic Database class that has these functionalities as methods. Then based on the DB specified (cassandra, postgres, etc) we would choose the proper class and everything would use the class methods. Simple in principle.

We'd welcome any help on this one. I think if left to us, it will take a month to get done given our other priorities. If only nights and weekends were longer 😅

**Reply by atsetilam** [reply]
Thank you @RamiAwar . I'll be contacting you via mail soon!

**Reply by RamiAwar** [reply]
Oh just join us on disco, much easier!

https://discord.com/invite/8NakmnVS
